FirebaseAppCheck/Sources/AppAttestProvider/FIRAppAttestProvider.m

```diff
@@ -118,18 +126,51 @@ - (nullable instancetype)initWithApp:(FIRApp *)app {
 #pragma mark - FIRAppCheckProvider
 
 - (void)getTokenWithCompletion:(void (^)(FIRAppCheckToken *_Nullable, NSError *_Nullable))handler {
+  [self getToken]
+      // Call the handler with the result.
+      .then(^FBLPromise *(FIRAppCheckToken *token) {
+        handler(token, nil);
+        return nil;
+      })
+      .catch(^(NSError *error) {
+        handler(nil, error);
+      });
+}
+
+- (FBLPromise<FIRAppCheckToken *> *)getToken {
+  // Check attestation state to decide on the next steps.
+  return [self attestationState].thenOn(self.queue, ^id(FIRAppAttestProviderState *attestState) {
+    switch (attestState.state) {
+      case FIRAppAttestAttestationStateUnsupported:
+        return attestState.appAttestUnsupportedError;
+        break;
+
+      case FIRAppAttestAttestationStateSupportedInitial:
+      case FIRAppAttestAttestationStateKeyGenerated:
+        // Initial handshake is required for both the "initial" and the "key generated" states.
```

Comment:
@ncooke3 In relation to the [comment](https://github.com/firebase/firebase-ios-sdk/pull/8032/files#r626776183). I updated this comment with a bit more details. Do you think it is clear enough or you would prefer to add see a separate comment for `FIRAppAttestAttestationStateSupportedInitial` case?

Reply:
Oh i see. Re-read and i now understand. That LGTM– sorry for confusion!

Reply:
No problem!